lib/matplotlib/collections.py

```diff
@@ -219,22 +220,23 @@ def get_transforms(self):
 
     def get_offset_transform(self):
         """Return the `.Transform` instance used by this artist offset."""
-        if self._transOffset is None:
-            self._transOffset = transforms.IdentityTransform()
-        elif (not isinstance(self._transOffset, transforms.Transform)
-              and hasattr(self._transOffset, '_as_mpl_transform')):
-            self._transOffset = self._transOffset._as_mpl_transform(self.axes)
-        return self._transOffset
+        if self._offset_trf is None:
+            self._offset_trf = transforms.IdentityTransform()
+        elif (not isinstance(self._offset_trf, transforms.Transform)
+              and hasattr(self._offset_trf, '_as_mpl_transform')):
+            self._offset_trf = self._offset_trf._as_mpl_transform(self.axes)
```

Comment:
I wonder why this sort of thing didn't happen in the setter? Is it some kind of optimization to avoid calling `_as_mpl_transform`?

If it were in the setter, you wouldn't need `offset_trf` because they'd all fit in a line even with the full name.

Reply:
You'll have to ask @pelson who added this in https://github.com/matplotlib/matplotlib/pull/1090.  I guess that in theory the pseudo-transform (the object with an _as_mpl_transform method) could be mutable and you may want the later mutations to be reflected at draw time?

Reply:
I can't remember a strong reason for it (though there might have been one). Perhaps there is some delay in getting an ``axes`` on a collection, and you can only guarantee it exists when you ``get_offset_transform``?

Certainly worth exploring though.

Reply:
Let's defer this to another issue, perhaps?

Reply:
OK, I was mostly trying to avoid the `offset_trf` abbreviation entirely, and not go into performance details just now.

Reply:
Fair point, I renamed to _offset_transform.

Reply:
I'd go as far as preferring `offset_transform` also for local variables (unless you have really bad wrapping issues). `trf` does not speak to me at all.

Reply:
But we already use trf for local variables if quite a few other places, and I'd rather try to establish it as a standard abbreviation (transform is quite a mouthful).  Or we can try to standardize on trans, if you prefer?

Reply:
I find `trans` slightly better than `trf`. I agree that a standard abbreviation is desireable.